scilpy/tracking/tracker.py

```diff
@@ -300,12 +300,20 @@ def _get_streamlines(self, chunk_id):
             line = self._get_line_both_directions(seed)
 
             if line is not None:
+                streamline = np.array(line, dtype='float32')
+
                 if self.compression_th and self.compression_th > 0:
-                    streamlines.append(
-                        compress_streamlines(np.array(line, dtype='float32'),
-                                             self.compression_th))
-                else:
-                    streamlines.append((np.array(line, dtype='float32')))
+                    # Compressing. Threshold is in mm. Verifying space.
+                    if self.space == Space.VOX:
+                        # Equivalent of sft.to_voxmm:
+                        streamline *= self.seed_generator.voxres
+                        compress_streamlines(streamline, self.compression_th)
```

Comment:
You could just transform the compression threshold to voxel space by dividing by voxel resolution, no?

Reply:
I'm not sure how compression works internally...

Reply:
I think the working principle is if the shortest distance between some point and the segment connecting its two immediate neighbours is smaller than the threshold, it is removed from the compressed streamline. The space in which the compression is done should not matter, see docstring [here](https://github.com/dipy/dipy/blob/08b89939a886f0b4a38d3eebc66cb0d1aae7a4fc/dipy/tracking/streamlinespeed.pyx#L540).

Reply:
Ok. However, here the voxel resolution is a 3D vector. If the resolution is not iso, we cannot do what you suggest. So I think this should still be the best way.

Reply:
ouais ok, true for non-iso.